TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -303,6 +303,21 @@ public class VehicleBehaviorManager : AbstractCustomManager, IVehicleBehaviorMan
 					//}
 #endif
 
+					// Check if turning in the preferred direction, and if turning while it's red is allowed
+					if (stopCar && sqrVelocity <= TrafficPriorityManager.MAX_SQR_STOP_VELOCITY && JunctionRestrictionsManager.Instance.IsTurnOnRedAllowed(prevPos.m_segment, isTargetStartNode)) {
+						SegmentGeometry currentSegGeo = SegmentGeometry.Get(prevPos.m_segment);
+						SegmentEndGeometry currentSegEndGeo = currentSegGeo.GetEnd(targetNodeId);
+						ArrowDirection targetDir = currentSegEndGeo.GetDirection(position.m_segment);
+						bool lhd = Services.SimulationService.LeftHandDrive;
+						if (lhd && targetDir == ArrowDirection.Left || !lhd && targetDir == ArrowDirection.Right) {
+#if DEBUG
+							if (debug)
+								Log._Debug($"VehicleBehaviorManager.MayChangeSegment({frontVehicleId}): Vehicle may turn on red to target segment {position.m_segment}, lane {position.m_lane}");
+#endif
+							stopCar = false;
+						}
+					}
+
```

Comment:
I clean up and moved this part up. If `stopCar = true` we got a red light. We only check if the next segment is either right (RHD) or left (LHD) relative to the previous segment and set `stopCar` to `false` in this case. The remaining logic will do the rest.
The upcoming lines handle the priority feature for timed traffic lights. Unlike before, if turn-on-red is allowed, the priority check will be executed.
At simulation time we don't have and should not have to perform complex checks (former logic in SegmentGeometry) whether the 'turn-on-red' feature may be applied at this node. The JunctionRestrictionManager is notified as soon as the segment geometry changes.

Reply:
Yup, priority rules were the missing part here